### PR TITLE
integrate FDB.1 with generic database server

### PR DESCRIPTION
**Pull request by jni2000**
Please review and comment on this integration of FDB.1 and generic database server. Other FDB.x and WDB.x will follow when this one is approved. Thanks.

### CODE REVIEW

**Comment by jni2000** [comment]
Kate and Taran as well as other reviewers

Please can you take a quick review on this FDB.1 and database server integration? We will submit the others when this one is approved.

Thanks.

File: src/harness/testcases/WINNF_FT_S_FDB_testcase.py [hunk 1]
```diff
@@ -162,9 +172,30 @@ def test_WINNF_FT_S_FDB_1(self, config_filename):
...
     # Step 2: Create exclusion zone database which contains the
     # CBSD location 'X' or is within 50 meters of the CBSD location 'X'.
+    # Create fake database server
+    fake_database_server = DatabaseServer(
+                              config['fakeDatabaseInfo']['name'],
+                              config['fakeDatabaseInfo']['hostName'],
+                              config['fakeDatabaseInfo']['port'],
+                              cert_file='certs/server.cert',
+                              key_file='certs/server.key',
+                              ca_cert_file='certs/ca.cert')
```
[the remark this answers is not in the document]
Comment (TaranFurnell) [reply src/harness/testcases/WINNF_FT_S_FDB_testcase.py]: Alternatively we go the other way and set defaults in the database object. As the certs for this don't really matter, just that they exist. What do you think?

File: src/harness/testcases/WINNF_FT_S_FDB_testcase.py [hunk 2]
```diff
@@ -135,9 +146,8 @@ def generate_FDB_1_default_config(self, filename):
             [grant_g1_a, grant_g1_b],
             [grant_g2_a, grant_g2_b],
             [grant_g3_a, grant_g3_b]],
-        'conditionalRegistrationData': conditionals
-        # TODO
-        # Need to add data base configurations.
+        'conditionalRegistrationData': conditionals,
+        'fakeDatabaseInfo': fake_database_config
```
[the remark this answers is not in the document]
Comment (jni2000) [reply src/harness/testcases/WINNF_FT_S_FDB_testcase.py]: Updated as per the comment.

File: src/harness/testcases/WINNF_FT_S_FDB_testcase.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jni2000) [reply src/harness/testcases/WINNF_FT_S_FDB_testcase.py]: Replaced the "database.py" file and updated the test case as per   the comment.

File: src/harness/testcases/WINNF_FT_S_FDB_testcase.py [hunk 3]
```diff
@@ -162,9 +172,30 @@ def test_WINNF_FT_S_FDB_1(self, config_filename):
                                    grant_request_g1,
                                    config['conditionalRegistrationData'])
 
-    # TODO
     # Step 2: Create exclusion zone database which contains the
     # CBSD location 'X' or is within 50 meters of the CBSD location 'X'.
+    # Create fake database server
+    fake_database_server = DatabaseServer(
+                              config['fakeDatabaseInfo']['name'],
```
[the remark this answers is not in the document]
Comment (jni2000) [reply src/harness/testcases/WINNF_FT_S_FDB_testcase.py]: Updated as per the comment.

File: src/harness/testcases/WINNF_FT_S_FDB_testcase.py [hunk 4]
```diff
@@ -162,9 +172,30 @@ def test_WINNF_FT_S_FDB_1(self, config_filename):
...
+    
+    # Start fake database server
+    fake_database_server.start()
+
+    # Inject the exclusion zone database URL into the SAS UUT
+    self._sas_admin.InjectDatabaseUrl("https://" +
+                                      config['fakeDatabaseInfo']['hostName'] +
+                                      ":" +
+                                      str(config['fakeDatabaseInfo']['port'])+
+                                      config['fakeDatabaseInfo']['fileUrl'])
```
[the remark this answers is not in the document]
Comment (jni2000) [reply src/harness/testcases/WINNF_FT_S_FDB_testcase.py]: Updated as per the comment.

**Comment by jni2000** [comment]
Taran

We have addressed all your latest review comments and made necessary changes, please take a look. We appreciate if you can respond to us on this one as soon as possible, so that we can duplicate the logic to FDB.x and WDB.x test cases to meet our 5/7 deliver goal.

Thanks.

File: src/harness/testcases/WINNF_FT_S_FDB_testcase.py [hunk 5]
```diff
@@ -162,9 +171,23 @@ def test_WINNF_FT_S_FDB_1(self, config_filename):
...
-    # TODO
     # Step 2: Create exclusion zone database which contains the
     # CBSD location 'X' or is within 50 meters of the CBSD location 'X'.
+    # Create fake database server
+    fake_database_server = DatabaseServer("Exclusion Zone Database",
+                                          config['exclusionZoneDatabaseConfig']['hostName'],
+                                          config['exclusionZoneDatabaseConfig']['port'])
+    
+    # Start fake database server
+    fake_database_server.start()
```
Comment (TaranFurnell) [inline src/harness/testcases/WINNF_FT_S_FDB_testcase.py]: nit: ```exclusion_zone_database_server```?

**Comment by TaranFurnell** [comment]
Note: I can't check the KML files atm. I'm ok with accepting this for now and correcting problems afterwards. So we can get unblocked on the other testcases.

**Comment by jni2000** [comment]
Thanks.

**Pull request merged**